httpclient/client.go

```diff
@@ -219,23 +232,52 @@ func (jc *HttpClient) ReadRemoteFile(downloadPath string, httpClientsDetails htt
 }
 
 func (jc *HttpClient) DownloadFile(downloadFileDetails *DownloadFileDetails, logMsgPrefix string, httpClientsDetails httputils.HttpClientDetails, retries int, isExplode bool) (*http.Response, error) {
-	resp, _, err := jc.downloadFile(downloadFileDetails, logMsgPrefix, true, httpClientsDetails, retries, isExplode)
+	resp, _, err := jc.downloadFileWithRetries(downloadFileDetails, logMsgPrefix, true, httpClientsDetails, retries, isExplode)
 	return resp, err
 }
 
 func (jc *HttpClient) DownloadFileNoRedirect(downloadPath, localPath, fileName string, httpClientsDetails httputils.HttpClientDetails) (*http.Response, string, error) {
 	downloadFileDetails := &DownloadFileDetails{DownloadPath: downloadPath, LocalPath: localPath, FileName: fileName}
-	return jc.downloadFile(downloadFileDetails, "", false, httpClientsDetails, 0, false)
+	return jc.downloadFile(downloadFileDetails, "", false, httpClientsDetails, false)
 }
 
-func (jc *HttpClient) downloadFile(downloadFileDetails *DownloadFileDetails, logMsgPrefix string, followRedirect bool,
+func (jc *HttpClient) downloadFileWithRetries(downloadFileDetails *DownloadFileDetails, logMsgPrefix string, followRedirect bool,
```

Comment:
Since I think all downloads should be with retries, I think we should not add "...WithRetries" to the name of any of the download methods.
The naming convention of the upload methods seems to be good, so maybe we should use the same naming convention of the download methods as well.